### ISSUE TITLE
Music Widget Features

### ISSUE THREAD

**Issue by mattingg** [ClutchEngineering/Sidecar#40]
It would be quite handy if i could tap the music at the bottom of the screen & have it go right to the music app which is playing the current song.

Is it possible to make the layout of the music widget more customise-able? I’d like to see the track time/ scrubbing bar underneath just like on the CarPlay homescreen, but can totally understand not everyone would want that

Another small consistency thing- Whenever a song plays that doesn’t have cover art, it won’t show any cover art, unlike on the CarPlay homescreen it replaces it with a ‘🎵’ icon. Is it possible to have this icon shown for these songs/an option to toggle it?

**Comment by jverkoey** [comment]
> It would be quite handy if i could tap the music at the bottom of the screen & have it go right to the music app which is playing the current song.

Widgets unfortunately can't be interactive in CarPlay due to limitations in how touch events are forwarded to the app. The only non-map interactions I'm able to react to in Sidecar are the four buttons on the side and buttons in the drop-down navigation bar at the top.

That being said, what is your intent in this described interaction? Are you wanting to be able to control the music in some manner, e.g. favoriting or pausing or something else?

> Is it possible to make the layout of the music widget more customise-able? I’d like to see the track time/ scrubbing bar underneath just like on the CarPlay homescreen, but can totally understand not everyone would want that

For similar reasons as above, an interactive scrubber is not possible. For track time, are you wanting to see the time remaining/progressed or the track duration, or both?

> Whenever a song plays that doesn’t have cover art, it won’t show any cover art, unlike on the CarPlay homescreen it replaces it with a ‘🎵’ icon. Is it possible to have this icon shown for these songs/an option to toggle it?

Ah! This is a bug; thank you for reminding me because I had noticed that for one of my tracks that doesn't have album art 😅 I'll prioritize fixing this for the next release.

**Comment by mattingg** [comment]
Yeah, i was thinking about pausing, favouriting tracks and also being able to change playlist, that’s exactly why i’d asked about being able to tap through to the designated music player 😌 

Yeah i’d like to see both the remaining/progressed track time and track duration!

Amazing i’ll be on the lookout for next release😎 Also, thank you for replying so promptly!

**Comment by jverkoey** [comment]
<img width="390" alt="Screenshot 2024-11-10 at 10 10 23 PM" src="https://github.com/user-attachments/assets/73e8003e-97a9-464f-9c2e-a5c1b581c173">

Fixed :)

Also handling empty artist names now too to boot!

**Comment by jverkoey** [comment]
> pausing

Added as a feature request for one of the four buttons you can configure.

> favouriting tracks

Apple Music doesn't seem to have an API for this yet sadly. I've filed FB15757413 to Apple with a suggestion to add this as an API to MusicKit

> being able to change playlist

I may look into this; it's something I could in theory add to the top navigation bar as a drill-down menu, similar to the OBD menu you can currently access from the top navigation bar. Filed as a feature request.

> i’d like to see both the remaining/progressed track time and track duration!

I'll think about how I might do this in a non-distracting way. Would you be ok if it was more like a progress bar being drawn around album art? I think this would be the ideal balance between not changing too frequently while also allowing a quick glance at the current progress in the track. Or do you specifically want to understand the amount of time remaining as a numerical quantity?

**Comment by mattingg** [comment]
Amazing. The progress bar around the cover art is a good idea, i like the sound of that. Maybe you could add a little customisation option to allow the user the change the colour of the progression bar also? Could help make it more visible to certain people😁

**Comment by studiograham** [comment]
The music playing should be “now playing” it doesn’t support podcasts or audiobooks.

**Comment by CaiseLava** [comment]
The now playing doesn’t support spotify for me at least,

**Comment by Nimijneb** [comment]
Want to add on that support for overcast podcast app would be great.

**Comment by s-sarkhosh** [comment]
I want to also second Overcast support. I listen to podcasts on my morning drive and music in the afternoon and it’s a bummer I have to switch out of the app to go back to Overcast to see what’s playing.
